### PR TITLE
[dagster-airlift] materialization accumulation refactor

### PR DESCRIPTION
**Pull request by dpeng817**
## Summary & Motivation
Remove materialization retrieval from what will soon be a pluggable implementation function.

## How I Tested These Changes
Existing tests

## Changelog
NOCHANGELOG

### CODE REVIEW

**Comment by dpeng817** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24830?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24831** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24831?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24830** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24830?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24829** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24829?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24828** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24828?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/dagster_airlift/core/sensor.py [hunk 1]
```diff
@@ -157,9 +157,7 @@ def materializations_and_requests_from_batch_iter(
         offset=offset,
     )
     for i, dag_run in enumerate(runs):
-        mats: List[AssetMaterialization] = []
-        mats.extend(materializations_for_dag_run(dag_run, airflow_data))
-        for task_run in airflow_instance.get_task_instance_batch(
+        task_instances = airflow_instance.get_task_instance_batch(
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/core/sensor.py:160]: doesn't this get overidden N times?

**Review by schrockn — COMMENTED** [review]
A bit curious about the PR summary. You mention "hot path" but I don't see how this really moves any core functionality. Seems like pure refactor.

File: examples/experimental/dagster-airlift/dagster_airlift/core/sensor.py (hunk 1 above)
Comment (dpeng817) [reply examples/experimental/dagster-airlift/dagster_airlift/core/sensor.py:169]: task instances are scoped to dag run (a task instance is just a run of task), so we need to retrieve them on a per-dag-run basis

Comment (schrockn) [reply examples/experimental/dagster-airlift/dagster_airlift/core/sensor.py:169]: well that makes perfect sense doesn't it

**Review by schrockn — APPROVED** [review]

**Pull request merged**